### PR TITLE
Add range-based iteration

### PR DESCRIPTION
**Pull request by mherzl**
create `iter_range`

add tests from #290

### CODE REVIEW

File: roaring/src/bitmap/iter.rs [hunk 1]
```diff
@@ -190,6 +191,10 @@ impl Iter<'_> {
     pub fn advance_back_to(&mut self, n: u32) {
         advance_back_to_impl(n, &mut self.front, &mut self.containers, &mut self.back);
     }
+
+    pub fn empty() -> Iter<'static> {
```
Comment (Kerollmops) [inline roaring/src/bitmap/iter.rs:195]: Would it be possible to add a bit of documentation to this method, please?

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you for the PR @mherzl,

I think I'll have to bump the minimal supported Rust version to v1.67 so that the CI passes again 🤔

**Comment by Kerollmops** [comment]
Sorry for that @mherzl but I will merge #297 as it was created earlier and there are more comments and discussions there. 

I'm sorry about this. Closing this one.
Have a nice day 🌹

**Comment by mherzl** [comment]
It's alright @Kerollmops. Maybe I'll come up with another feature to add at some point.

**Pull request closed without merging**